bgpd/bgp_route.c

```diff
@@ -11971,6 +11997,7 @@ DEFUN (no_bgp_distance,
 	bgp->distance_ebgp[afi][safi] = 0;
 	bgp->distance_ibgp[afi][safi] = 0;
 	bgp->distance_local[afi][safi] = 0;
+	bgp_announce_routes_distance_update(bgp);
```

Comment:
This is dead code

Reply:
Will fix this :+1:

Reply:
> This is dead code

looks ike it is needed.. here is the call flow bgp_zebra_announce_table() -> bgp_zebra_announce() --> bgp_distance_apply(), her it updated the default distance. There after the ZAPI_MESSAGE_DISTANCE flag is set and set to zebra inside bgp_zebra_announce()
Am i missing something ? :)

Reply:
@bisdhdh Quentin had in his mind that bgp_announce_routes_distance_update() won't do anything if distance_* variables are set to 0 due to:
```
if (bgp->distance_ibgp[afi][safi] && ...)
```

Reply:
> @bisdhdh Quentin had in his mind that bgp_announce_routes_distance_update() won't do anything if distance_* variables are set to 0 due to:
> 
> ```
> if (bgp->distance_ibgp[afi][safi] && ...)
> ```

lol ! so this was in the previous version of the code.. n i am searching in the lets one :P